integration-libs/opf/base/components/opf-cta/opf-cta-scripts/opf-cta-scripts.component.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { of, throwError } from 'rxjs';
+import { OpfCtaWrapperComponent } from './opf-cta-scripts.component';
+import { OpfCtaScriptsService } from './opf-cta-scripts.service';
+import createSpy = jasmine.createSpy;
+
+const mockHtmlsList = [
+  '<div  style="border-style: solid;text-align:center;border-radius:10px;align-content:center;background-color:yellow;color:black"><h2>Thanks for purchasing our great products</h2><h3>Please use promo code:<b>123abc</b> for your next purchase<h3></div><script>console.log(\'CTA Script #1 is running\')</script>',
+  '<div  style="border-style: solid;text-align:center;border-radius:10px;align-content:center;background-color:yellow;color:black"><h2>Thanks again for purchasing our great products</h2><h3>Please use promo code:<b>123abc</b> for your next purchase<h3></div><script>console.log(\'CTA Script #2 is running\')</script>',
+];
+const ctaElementSelector = 'cx-opf-cta-element';
+describe('OpfCtaScriptsComponent', () => {
+  let component: OpfCtaWrapperComponent;
+  let fixture: ComponentFixture<OpfCtaWrapperComponent>;
+  let opfCtaScriptsService: jasmine.SpyObj<OpfCtaScriptsService>;
+
+  beforeEach(() => {
+    opfCtaScriptsService = jasmine.createSpyObj('OpfCtaScriptsService', [
+      'getCtaHtmlslList',
+    ]);
+
+    TestBed.configureTestingModule({
+      declarations: [OpfCtaWrapperComponent],
+      providers: [
+        { provide: OpfCtaScriptsService, useValue: opfCtaScriptsService },
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    opfCtaScriptsService.getCtaHtmlslList.and.returnValue(of(mockHtmlsList));
+    fixture = TestBed.createComponent(OpfCtaWrapperComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should return Htmls list and display ctaButton elements', (done) => {
+    component.ctaHtmls$.subscribe((htmlList) => {
+      expect(htmlList[0]).toBeTruthy();
+      fixture.detectChanges();
+      expect(
+        fixture.nativeElement.querySelectorAll(ctaElementSelector).length
+      ).toEqual(2);
+      done();
+    });
+  });
+
+  it('should isError be true when error is thrown', (done) => {
+    opfCtaScriptsService.getCtaHtmlslList = createSpy().and.returnValue(
+      throwError('error')
+    );
+    fixture = TestBed.createComponent(OpfCtaWrapperComponent);
+    component = fixture.componentInstance;
```

Comment:
I haven't found any better solution to force  opfCtaScriptsService.getCtaHtmlslList fct generating a new value.
fixture and component needs to be recerated with 'it' block.
Tried with getCtaHtmlslList
I had another solution with Subject to mock getCtaHtmlslList fct, it is even more DIY causing subscriptions probs.